snakemake/deployment/docker.py

```diff
@@ -0,0 +1,119 @@
+__author__ = "Johannes Köster"
+__copyright__ = "Copyright 2015-2020, Johannes Köster"
+__email__ = "johannes.koester@uni-due.de"
+__license__ = "MIT"
+
+import os
+import re
+import requests
+
+
+class DockerContainer:
+    """A Docker Container is a thin wrapper to a Docker Registry API to
+       primarily determine if a container exists to be used for a workflow
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.uri = None
+        if self.exists():
+            self.uri = "docker://%s" % self.name
+
+    @property
+    def version(self):
```

Comment:
The version property can take a tag (e.g., latest) or a container hash (specified with `@` as a version). For snakemake wrapper containers, the version will always be the hash of the conda environment versions. For external wrappers, or wrappers otherwise given a docker:// uri, the version can be a tag or hash.